File: src/main/java/no/novari/kafka/topic/name/exceptions/MissingTopicNameParameterException.java
package no.novari.kafka.topic.name.exceptions;

public class MissingTopicNameParameterException extends RuntimeException {

    private MissingTopicNameParameterException(String parameterName, String state) {
        super("Required parameter '" + parameterName + "' is " + state);
    }

    public static MissingTopicNameParameterException notDefined(String parameterName) {
        return new MissingTopicNameParameterException(parameterName, "not defined");
    }

    public static MissingTopicNameParameterException blank(String parameterName) {
        return new MissingTopicNameParameterException(parameterName, "blank");

    }

}

File: src/main/kotlin/no/novari/kafka/topic/name/exceptions/MissingTopicNameParameterException.kt
package no.novari.kafka.topic.name.exceptions

class MissingTopicNameParameterException private constructor(
    parameterName: String?,
    state: String,
) : RuntimeException("Required parameter '$parameterName' is $state") {
    companion object {
        @JvmStatic
        fun notDefined(parameterName: String?): MissingTopicNameParameterException =
            MissingTopicNameParameterException(parameterName, "not defined")

        @JvmStatic
        fun blank(parameterName: String?): MissingTopicNameParameterException =
            MissingTopicNameParameterException(parameterName, "blank")
    }
}
